src/features/import-map.js

```diff
@@ -41,7 +41,8 @@ systemJSPrototype.prepareImport = function () {
 
 systemJSPrototype.resolve = function (id, parentUrl) {
   parentUrl = parentUrl || baseUrl;
-  return resolveImportMap(importMap, resolveIfNotPlainOrUrl(id, parentUrl) || id, parentUrl) || throwUnresolved(id, parentUrl);
+  const map = hasDocument ? importMap : (this._importMap || emptyMap());
+  return resolveImportMap(map, resolveIfNotPlainOrUrl(id, parentUrl) || id, parentUrl) || throwUnresolved(id, parentUrl);
 };
```

Comment:
Since we need to scope the import map to the System instance, I would suggest we use a Symbol for this now, just like we use symbols in core.

Then always use the internally scoped import map without the `hasDocument` check.

This will also ensure that new instantiations of the System loader in browsers also support custom import maps.

Reply:
I've also included this in the most recent commit.